src/main/java/org/vanilladb/core/storage/record/RecordFile.java

```diff
@@ -398,15 +413,8 @@ private void appendBlock() {
 	}
 
 	private FileHeaderPage openHeaderForModification() {
-		// acquires exclusive access to the header
-		if (!isTempTable()) {
-			fhpWaitCount.incrementAndGet();
-			tx.concurrencyMgr().lockRecordFileHeader(headerBlk);
-			fhpReleaseCount.incrementAndGet();
-			fhpWaitCount.decrementAndGet();
-
-		}
-		return new FileHeaderPage(fileName, tx);
+		FileHeaderPage fhp = new FileHeaderPage(fileName, tx);
+		return fhp;

```

Comment:
It seems like there is no corresponding modification for `delete()` and `insert(RecordId)`. We better add a warning like `TODO` or `XXX` here.

Reply:
I've modified both to reduce the future works